src/EditorFeatures/Core/LanguageServer/Handlers/Completion/CompletionHandler.cs

```diff
@@ -229,20 +193,19 @@ bool IsValidTriggerCharacterForDocument(Document document, char triggerCharacter
                 var completeDisplayText = stringBuilder.ToString();
                 stringBuilder.Clear();
 
-                var completionItem = new TCompletionItem
-                {
-                    Label = completeDisplayText,
-                    SortText = item.SortText,
-                    FilterText = item.FilterText,
-                    Kind = GetCompletionKind(item.Tags),
-                    Data = completionResolveData,
-                    Preselect = ShouldItemBePreselected(item),
-                };
+                var completionItem = supportsVSExtensions ? new LSP.VSInternalCompletionItem() : new LSP.CompletionItem();
+                completionItem.Label = completeDisplayText;
+                completionItem.SortText = item.SortText;
+                completionItem.FilterText = item.FilterText;
+                completionItem.Kind = GetCompletionKind(item.Tags);
+                completionItem.Data = completionResolveData;
+                completionItem.Preselect = ShouldItemBePreselected(item);
 
                 // Complex text edits (e.g. override and partial method completions) are always populated in the
                 // resolve handler, so we leave both TextEdit and InsertText unpopulated in these cases.
-                if (item.IsComplexTextEdit)
+                if (item.IsComplexTextEdit && completionItem is LSP.VSInternalCompletionItem vsItem)
```

Comment:
Are ComplexTextEdit's only supported in VS? I would have thought `LSP.VSInternalCompletionItem` should be it's own check but perhaps I don't understand the scenario.

Reply:
complex text edits exist outside of VS, but the capability to resolve on commit is VS only.  So it doesn't make sense to do this block outside of VS